src/Sentry/Platforms/Native/UnmanagedHttpContent.cs

```diff
@@ -0,0 +1,63 @@
+using Sentry.Extensibility;
+
+namespace Sentry.Native;
+
+internal class UnmanagedHttpContent : SerializableHttpContent
+{
+    private readonly IntPtr _content;
+    private readonly int _length = 0;
+    private readonly IDiagnosticLogger? _logger;
+
+    public UnmanagedHttpContent(IntPtr content, int length, IDiagnosticLogger? logger)
+    {
+        _content = content;
+        _length = length;
+        _logger = logger;
+    }
+
+    protected override Task SerializeToStreamAsync(Stream stream, TransportContext? context)
```

Comment:
question: overload

I believe there is alsp a virtual overload of `SerializeToStreamAsync` that takes a `CancellationToken cancellationToken`.
Would we like to override this overload as well, so that we also support asynchronous cancellation?

Reply:
Supposedly, it needs some `NET5_0_OR_GREATER` guards to retain compatibility with older targets. Do you want to do that in this PR, or is that something that could be done in `SerializableHttpContent` as a separate step?